plugin/core/handlers.py

```diff
@@ -17,9 +17,19 @@ def config(self) -> ClientConfig:
 
     @classmethod
     def instantiate_all(cls) -> 'List[LanguageHandler]':
-        return list(
-            instantiate(c) for c in cls.__subclasses__()
-            if issubclass(c, LanguageHandler))
+        subclasses = []  # type: List[Type[LanguageHandler]]
+        cls.get_final_subclasses(cls.__subclasses__(), subclasses)
+        return list(instantiate(c) for c in subclasses)
+
+    @classmethod
+    def get_final_subclasses(cls, superclasses: 'List[Type[LanguageHandler]]',
```

Comment:
The name of this variable is confusing. You're passing the subcasses / derived_classes, not the superclasses / base_classes.